### ISSUE TITLE
3.1.0.2 breaks mc

### ISSUE THREAD

**Issue by greenaar** [mintty/wsltty#217]
mc (midnight commander) under WSL ubuntu 18.04 hangs permanently on 3.1.0.2 (with extreme cpu usage showing for wslbridge2).

Works on 3.1.0.

This is true for both the i686 and x64 versions, and confirmed on multiple systems (windows 10 pro).

**Comment by Biswa96** [comment]
No issue in my system, Windows 10 latest insider build.

**Comment by mintty** [comment]
This is an issue of cygwin 3.1.0 and has been discussed on the cygwin mailing list.
Seems to be fixed with latest cygwin. May need to bundle an update for wsltty.
(Will take a while due to holiday season...)

**Comment by Biswa96** [comment]
But that cygwin issue[1] was with mc in cygwin, here mc in WSL. Are these cases related?

[1]: https://cygwin.com/ml/cygwin/2019-12/msg00173.html

**Comment by Biswa96** [comment]
But that [cygwin issue][1] was with mc in cygwin, here mc in WSL. Are these cases related?

[1]: https://cygwin.com/ml/cygwin/2019-12/msg00173.html

**Comment by piotrzientarski** [comment]
I can confirm that - Midnight Commander does not work on version 3.1.0.2

I'm using windows 10 pro with Ubuntu 18.04 on WSL2.

**Comment by mintty** [comment]
Please update mc to upstream update 4.8.24 which fixes handling of window size changes.
It seems to solve the issue with cygwin mc, at least.

**Comment by karanikn** [comment]
mc not working 

wsltty-3.1.0.2-install-x86_64.exe on Windows 10 Pro 1909 Build: 18363.592
Microsoft Windows [Version 10.0.18363.592]
mc version>>   mc:amd64/bionic 3:4.8.19-1 uptodate

also i checked with
mc:amd64/bionic 3:4.8.23-1 uptodate

**Comment by mintty** [comment]
See my previous comment:
> update 4.8.24

**Comment by long-nguyenxuan** [comment]
@mintty  
I'm newbie with linux, did try Google but can't update to 4.8.24, so can you help me a bit? 
How do I force my Ubuntu 1804 to update `mc` to 4.8.24, my `mc` is 4.8.19 
tried `sudo apt update` many times but it tells me `All packages are up to date.`

**Comment by Biswa96** [comment]
@long-nguyenxuan Sir, you are using Ubuntu 18.04 LTS version. Generally, packages in LTS version is not the latest one. You can use latest Ubuntu versions.

**Comment by long-nguyenxuan** [comment]
> @long-nguyenxuan Sir, you are using Ubuntu 18.04 LTS version. Generally, packages in LTS version is not the latest one. You can use latest Ubuntu versions.

thanks, let me try another one

**Comment by mintty** [comment]
4.8.24 is quite new. If Ubuntu does not yet offer it, you might compile it yourself. Should that not be straight-forward, though, we cannot support building external packages here.

**Comment by long-nguyenxuan** [comment]
I understood, thanks for the clarification.

**Comment by raiden-dev** [comment]
Not working on debian stable (buster) host connected with ssh using wsltty 3.1.0.2 (win10 build 19559), mc 4.8.24 installed from testing (bullseye) repo.

```
$ mc --version
GNU Midnight Commander 4.8.24
Built with GLib 2.62.4
Using the S-Lang library with terminfo database
With builtin Editor and Aspell support
With subshell support as default
With support for background operations
With mouse support on xterm and Linux console
With support for X11 events
With internationalization support
With multiple codepages support
Virtual File Systems: cpiofs, tarfs, sfs, extfs, ext2undelfs, ftpfs, sftpfs, fish
Data types: char: 8; int: 32; long: 64; void *: 64; size_t: 64; off_t: 64;
```

Here's [dump](https://pastebin.com/N5u94Qvb) of mc screen that hangs wsltty.

**Comment by mintty** [comment]
Thank you very much for the test case (the dump log file).
It makes terminal output sort of hang, however not terminal interaction. Type Ctrl+d after the cat and mintty will terminate normally.
I've further narrowed the occurrence of the "hang" to two conditions, however the picture does not result in sufficient diagnostic information to find the actual cause.
It occurs when the "push DEC state" sequence is output: \e[?1001s.
It occurs with cygwin 3.1.0 (packaged in wsltty) but not if the cygwin dll is replaced with 3.1.2.
While I'm still puzzled how and why this happens, apparently it can be fixed by updating cygwin to 3.1.2, so I'll bundle a fresh wsltty package.

**Comment by mintty** [comment]
This was fixed in cygwin commit fe512b2b12a2cea8393d14f038dc3914b1bf3f60, as indicated in my first comment. Sorry I forgot to take care after holidays...
Maybe it's okay now to wait another few days for cygwin release 3.1.3?

**Comment by raiden-dev** [comment]
> Maybe it's okay now to wait another few days for cygwin release 3.1.3?

Sure, not a big problem for me. Thanks for your work 👍

**Comment by raiden-dev** [comment]
Hey, @mintty! It seems like cygwin 3.1.3 (and 3.1.4) has been released. Could you build a new wsltty release yet?

**Comment by mintty** [comment]
Released 3.1.4, including cygwin 3.1.5.
